modules/xfem/src/base/XFEM.C

```diff
@@ -150,6 +152,27 @@ XFEM::clearStateMarkedElems()
   _state_marked_elem_sides.clear();
 }
 
+void
+XFEM::addGeomMarkedElem2D(const unsigned int elem_id, const Xfem::GeomMarkedElemInfo2D geom_info)
+{
+  Elem * elem = _mesh->elem(elem_id);
```

Comment:
Needs protection for distributed mesh.

Reply:
I added this check in another place (XFEM::update())